client-js/app/elements/labrad-grapher.ts

```diff
@@ -35,30 +52,46 @@ export class LabradGrapher extends polymer.Base {
     this.showStars = false;
     this.showTrash = false;
     this.updateButtons();
+
+    this.initializeListItems_(
+      this.path,
+      this.dirs,
+      this.datasets);
+    this.applyListFilter_();
   }
 
   starClicked() {
     this.showStars = !this.showStars;
     if (this.showStars) this.showTrash = false;
     this.updateButtons();
-    this.kick++;
+    this.applyListFilter_();
   }
 
   trashClicked() {
```

Comment:
I know you didn't name this originally, but ```trashClicked()``` and ```starClicked()``` could use more descriptive function names. For a quick read through I can't tell if this is the function that filters the list or tags the datasets.